### PR TITLE
Add camera to pioneer2dx_noplugin

### PR DESCRIPTION
**Pull request by Jialn**
Add a 120 degree wide angle, 128x128 resolution camera to pioneer2dx_noplugin. 
Remove the old camera in world file. 
In simple_navigation env, camera scope name is also changed to the new camera's name.

### CODE REVIEW

File: python/social_bot/envs/simple_navigation.py [hunk 1]
```diff
@@ -155,7 +155,8 @@ def reset(self):
         return obs
 
     def get_camera_observation(self):
-        image = self._agent.get_camera_observation("camera")
+        image = self._agent.get_camera_observation(
+            "default::pioneer2dx_noplugin::camera::camera")
```
Comment (emailweixu) [inline python/social_bot/envs/simple_navigation.py]: For some unknown reason: I got the following errors on my machine (Ubuntu 18.04, Gazebo 10):
[Msg] Waiting for master.
[Msg] Connected to gazebo master @ http://127.0.0.1:11345
[Msg] Publicized address: 10.40.0.7
[Err] [LogRecord.cc:158] LogRecord has not been initialized.
[Err] [SensorManager.cc:296] Unable to create sensor of type[pioneer2dx_noplugin::camera]
[Wrn] [msgs.cc:1832] Conversion of sensor type[pioneer2dx_noplugin::camera] not supported.
 joint name: pioneer2dx::pioneer2dx_noplugin::camera_joint
 joint name: pioneer2dx::pioneer2dx_noplugin::left_wheel_hinge
 joint name: pioneer2dx::pioneer2dx_noplugin::right_wheel_hinge
INFO:social_bot.envs.simple_navigation:joint names: ['pioneer2dx::pioneer2dx_noplugin::camera_joint', 'pioneer2dx::pioneer2dx_noplugin::left_wheel_hinge', 'pioneer2dx::pioneer2dx_noplugin::right_wheel_hinge']
 joint name: pioneer2dx::pioneer2dx_noplugin::camera_joint
 joint name: pioneer2dx::pioneer2dx_noplugin::left_wheel_hinge
 joint name: pioneer2dx::pioneer2dx_noplugin::right_wheel_hinge
unable to find sensor: default::pioneer2dx_noplugin::camera::camera
Segmentation fault (core dumped)

Comment (hnyu) [reply python/social_bot/envs/simple_navigation.py]: I also have this same problem on my Ubuntu machine.

File: python/social_bot/models/pioneer2dx_noplugin/model.sdf [hunk 2]
```diff
@@ -62,6 +62,32 @@
         </material>
       </visual>
     </link>
+    <link name="camera">
```
Comment (emailweixu) [inline python/social_bot/models/pioneer2dx_noplugin/model.sdf]: It seems that you cannot use "camera" here. Changing back to "link" can work.

File: python/social_bot/models/pioneer2dx_noplugin/model.sdf [hunk 3]
```diff
@@ -154,6 +180,15 @@
         </material>
       </visual>
     </link>
+    <joint type="fixed" name="camera_joint">
```
Comment (emailweixu) [inline python/social_bot/models/pioneer2dx_noplugin/model.sdf]: This change make simple_navigation fail too. As simple_navigation assumes 3 control variables. The third one is the camera rotation.

File: python/social_bot/envs/simple_navigation.py (hunk 1 above)
Comment (Jialn) [reply python/social_bot/envs/simple_navigation.py]: Fixed.

File: python/social_bot/models/pioneer2dx_noplugin/model.sdf (hunk 2 above)
Comment (Jialn) [reply python/social_bot/models/pioneer2dx_noplugin/model.sdf]: Yes. Change it to "camera_link".

File: python/social_bot/models/pioneer2dx_noplugin/model.sdf (hunk 3 above)
Comment (Jialn) [reply python/social_bot/models/pioneer2dx_noplugin/model.sdf]: Change back to revolute

Comment (Jialn) [reply python/social_bot/models/pioneer2dx_noplugin/model.sdf]: Considering the observation does not include state of camera joint, would rotating the camera confuse the agent? For example, 180 degree rotated camera gives an image meaning an opposite  left & right.
As a 120 degree camera is wide enough, it's easy for the agent to rotate itself by the 2 wheels to seek the goal into its view. Maybe we can just remove the 3rd control variable and set the joint to fixed.
Also, we have to do some extra work to fix the joint when inserting pioneer as an agent to other environment that does not need a rotating camera.

Comment (emailweixu) [reply python/social_bot/models/pioneer2dx_noplugin/model.sdf]: It's okay to remove this control. But you need to change SimpleNavigation accordingly to make it still work.

File: python/social_bot/worlds/pioneer2dx_camera.world [hunk 4]
```diff
@@ -2,13 +2,13 @@
...
     <physics type="ode">
-      <max_step_size>0.001</max_step_size>
+      <max_step_size>0.005</max_step_size>
       <real_time_factor>1</real_time_factor>
       <real_time_update_rate>0</real_time_update_rate>
     </physics>
     <!-- Ground -->
     <include>
-      <uri>model://texture_ground_plane</uri>
+      <uri>model://ground_plane</uri>
```
Comment (emailweixu) [inline python/social_bot/worlds/pioneer2dx_camera.world]: why do you change to ground_plane from texture_ground_plane?

Comment (Jialn) [reply python/social_bot/worlds/pioneer2dx_camera.world]: Will change it back.
I don't know why but i have a problem loading the texture ground plane, gzclient takes
 quite a long time to display. (Waiting downloading model until timeout). I removed this for testing and forgot changing it back it before commit.

File: python/social_bot/models/pioneer2dx_noplugin/model.sdf (hunk 3 above)
Comment (Jialn) [reply python/social_bot/models/pioneer2dx_noplugin/model.sdf]: ok, I'll do that.

**Pull request merged**